applications/GeoMechanicsApplication/custom_retention/van_genuchten_law.cpp

```diff
@@ -51,9 +48,9 @@ double VanGenuchtenLaw::CalculateEffectiveSaturation(Parameters& rParameters) co
 {
     KRATOS_TRY
 
-    const auto&   rMaterialProperties = rParameters.GetMaterialProperties();
-    const double& satMax              = rMaterialProperties[SATURATED_SATURATION];
-    const double& satMin              = rMaterialProperties[RESIDUAL_SATURATION];
+    const auto& rMaterialProperties = rParameters.GetMaterialProperties();
+    const auto  satMax              = rMaterialProperties[SATURATED_SATURATION];
+    const auto  satMin              = rMaterialProperties[RESIDUAL_SATURATION];
```

Comment:
Just to be consistent with your other changes:
```suggestion
    const auto& r_material_properties = rParameters.GetMaterialProperties();
    const auto  sat_max               = rMaterialProperties[SATURATED_SATURATION];
    const auto  sat_min               = rMaterialProperties[RESIDUAL_SATURATION];
```

Reply:
done.